massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -138,14 +158,86 @@ impl SpeculativeLedger {
         // simulate crediting coins to destination address (if any)
         // note that to_addr can be the same as from_addr
         if let Some(to_addr) = to_addr {
-            let new_balance = changes
-                .get_balance_or_else(&to_addr, || self.get_balance(&to_addr))
-                .unwrap_or_default()
-                .checked_add(amount)
-                .ok_or_else(|| {
-                    ExecutionError::RuntimeError("overflow in to_addr balance".into())
-                })?;
-            changes.set_balance(to_addr, new_balance);
+            let old_balance = changes.get_balance_or_else(&to_addr, || self.get_balance(&to_addr));
+            match (old_balance, from_addr) {
+                // if `to_addr` exists we increase the balance
+                (Some(old_balance), _) => {
+                    let new_balance = old_balance.checked_add(amount).ok_or_else(|| {
+                        ExecutionError::RuntimeError("overflow in to_addr balance".into())
+                    })?;
+                    changes.set_balance(to_addr, new_balance);
+                }
+                // if `to_addr` doesn't exist but `from_addr` is defined. `from_addr` will create the address using the coins sent.
+                (None, Some(from_addr)) => {
+                    //TODO: Remove when stabilized
+                    debug!("Creating address {} from sender balance", to_addr);
+                    // Debit sender to create receiver address
+                    let address_storage_cost = self
```

Comment:
The whole ledger_cost_per_byte * ledger_entry_base_size should be added to the StorageCostsConstants structure under the name "ledger_entry_base_cost" and computed once-and-for-all in main with an expect() in case of overflow.

The same applies to the base datastore entry cost